projects/exui-common-lib/src/lib/components/generic-filter/generic-filter.component.ts

```diff
@@ -55,96 +58,237 @@ export class GenericFilterComponent {
+    const findPersonControl = this.form ? this.form.get('findPersonControl') : null;
+    this.form = this.fb.group({});
+    if (findPersonControl) {
+      // in order to maintain find person component, control needs to be kept
+      this.form.addControl('findPersonControl', findPersonControl);
+    }
+    for (const field of config.fields) {
+      if (field.type === 'checkbox') {
+        const formArray = this.buildCheckBoxFormArray(field, settings);
+        this.form.addControl(field.name, formArray);
+      } else {
+        const validators: ValidatorFn[] = [];
+        if (field.minSelected && field.minSelected > 0) {
+          validators.push(Validators.required);
         }
-
-        if (field.value.length > fieldConstraint.maxSelected) {
-          // display error for maximum selected
-          fieldConstraint.displayMaxSelectedError = true;
-          hasErrors = true;
-          this.filterService.givenErrors.next(fieldConstraint.maxSelectedError);
-          continue;
+        let defaultValue: string = '';
+        if (reset && config.cancelSetting) {
+          const cancelField = config.cancelSetting.fields.find((f) => f.name === field.name);
+          defaultValue = cancelField ? cancelField.value[0] : '';
+        } else if (settings && settings.fields) {
+          const lastSavedValue = settings.fields.find((f) => f.name === field.name);
+          defaultValue = lastSavedValue ? lastSavedValue.value[0] as unknown as string : '';
+        }
+        // if field is find-person build a form group;
+        if (field.type === 'find-person') {
+          const formGroup = new FormGroup({
+            domain: new FormControl(''),
+            email: new FormControl(defaultValue.hasOwnProperty('email') ? (defaultValue as any).email : '', validators),
+            id: new FormControl(''),
+            name: new FormControl(''),
+            knownAs: new FormControl(''),
+          });
+          this.form.addControl(field.name, formGroup);
+        } else {
+          const control = new FormControl(defaultValue, validators);
+          this.form.addControl(field.name, control);
         }
 
-        if (field.value.length < fieldConstraint.minSelected) {
-          // display error for minimum selected
-          fieldConstraint.displayMinSelectedError = true;
-          hasErrors = true;
-          this.filterService.givenErrors.next(fieldConstraint.minSelectedError);
-          continue;
+        // if field updates find person component set the initial domain
+        if (field.findPersonField) {
+          this.updateSpecificPerson(field, this.form);
         }
-        fieldConstraint.displayMinSelectedError = false;
-        fieldConstraint.displayMaxSelectedError = false;
       }
     }
-    return hasErrors;
+  }
+
+  private buildCheckBoxFormArray(field: FilterFieldConfig, settings: FilterSetting): FormArray {
+    const formArray = this.fb.array([], [
+      minSelectedValidator(field.minSelected),
```

Comment:
Are min and max selected always set for checkboxes? (I don't think they will be, plenty of scenarios where 0 up to all of them would be valid)